### PR TITLE
added wizard typing

### LINKED ISSUE #801: TypeScript for Wizard scene
Add TypeScript for Wizard scene

### PR DESCRIPTION
**Pull request by parsaalian**
# Description

This pull request adds support for wizard scenes in typescript.

Fixes #801

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

I have added a test file in `typings/stage.test.ts` which uses the new typing system and seems to work well with it.

**Test Configuration**:
* Node.js Version: v12.16.1
* Operating System: MacOSX 10.13.6

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation (no change in the documentation)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: typings/index.d.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /** @format */
 
-export { BaseScene, Stage } from './stage'
+export { BaseScene, WizardScene, Stage } from './stage'
```
Comment (wojpawlik) [inline typings/index.d.ts:3]: `WizardScene` is not exported from `telegraf.js`

Comment (parsaalian) [reply typings/index.d.ts:3]: Fixed in new commit

**Comment by wojpawlik** [comment]
Please move Wizard typings into a separate file, so it's easy to notice changes to scenes / stages.

File: typings/wizard.d.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
+/** @format */
+
+import { Composer, Middleware } from './composer'
+import { SceneContextOptions } from './stage'
+import { TelegrafContext } from './context'
+
+export interface WizardContext<TContext extends WizardContextMessageUpdate> {
+  ctx: TContext
+
+  options: SceneContextOptions
```
Comment (wojpawlik) [inline typings/wizard.d.ts:10]: https://github.com/telegraf/telegraf/blob/f5bb973015ec637cde26aeabf4c530926df1efe5/scenes/wizard/context.js#L2-L7 doesn't seem to have `options`.

**Comment by wojpawlik** [comment]
You know what, if @dotcypress allows, I'll be moving Telegraf to Typescript, see #1074.
You could then add type annotations directly to `scenes/**`, and TS would check them more reliably than I can.

**Comment by wojpawlik** [comment]
@parsaalian you can help by adding type annotations to scenes and `Stage`.

Git is not very smart at tracking renames, so first make a separate commit just for renaming the file(s) you wish to annotate, by using `git mv` and `git commit --no-verify`.

**Comment by foodornt** [comment]
are these changes applied yet? I can't import the WizardScene yet

**Pull request closed without merging**